### PR TITLE
How to code

### PR DESCRIPTION
**Pull request by patrikspiess**
I added a **How to Code** section and updated the TOC.
It's always a great pleasure to contribute in this marvelous project.

### CODE REVIEW

File: docs/source/developer/how_to_code.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+How to Code
+===========
+
+Before you start contributing in this project please read and accept our :ref:`C4`.
+
+Before you start coding always remember the *Zen of Python* by Tim Peters. If you don't know it just
+type :code:`import this` in an interactive Python interpreter.
+
+We strongly encourage you to use any Linux environment to code. The project was never tested on
```
Comment (lucmurer) [inline docs/source/developer/how_to_code.rst:13]: ".. Linux environment or WSL on Windows .."?

File: docs/source/developer/how_to_code.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+-   Use a formatter (we suggest `Black <https://github.com/psf/black>`_) to comply with our styling
+    guidelines.
+
+-   Use a linter (we suggest `pylint <https://github.com/pylint-dev/pylint>`_) for static code
+    analysis.
+
+**Line length**
+
+Maximum line length is 100 characters.
+
```
Comment (lucmurer) [inline docs/source/developer/how_to_code.rst:39]: ... These styles are enforced by a tox pipeline running `pylint` and `black` that also runs on every pull request on GitHub.

File: docs/source/developer/how_to_code.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+Use `Mypy <https://mypy.readthedocs.io/>`_ for static type checking. Always annotate your types.
+Mypy will remind you if you accidentally forget it. Because we annotate our types we do not add the
+types again in any docstrings.
+
+
+Testing
+-------
+
+Use `Pytest <https://docs.pytest.org/>`_ for code testing and test coverage measurement. We seek a
+coverage of at least 90%. You'll find more detailed information about writing tests in our
```
Comment (lucmurer) [inline docs/source/developer/how_to_code.rst:53]: -> We should bring this coverage of at least 90% to our tox pipeline too (currently still 80%).

File: docs/source/developer/how_to_code.rst [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+:ref:`HowToTest` documentation.
+
+
+Test Automation
+---------------
+
+Use `Tox <https://tox.wiki/en/latest/>`_ for test automation. Whenever you did some changes you wish
+to publish run :code:`tox` before any merge request. Every merge request which does not pass all
+tests wi be rejected.
```
Comment (lucmurer) [inline docs/source/developer/how_to_code.rst:62]: ... will ...

File: docs/source/developer/how_to_code.rst [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
```
Comment (lucmurer) [inline docs/source/developer/how_to_code.rst:63]: -> Information about Changelog is missing (keepachangelog)

File: docs/source/developer/how_to_release.rst [hunk 6]
```diff
@@ -38,7 +38,7 @@ documentation. These are the steps it goes through:
 ---------------
 
 The script `.tag_release.sh <https://github.com/migros/fotoobo/blob/main/.tag_release.sh>`_ just
-creates the release tag. This script has to be run the main repository `migros/fotoobo
+creates the release tag. This script has to be run at the main repository `migros/fotoobo
```
Comment (lucmurer) [inline docs/source/developer/how_to_release.rst:41]: ... in ...?

**Review by lucmurer — CHANGES_REQUESTED** [review]
Very good start, but I think there is still some things missing or can be optimized.

File: docs/source/developer/how_to_code.rst (hunk 1 above)
Comment (patrikspiess) [reply docs/source/developer/how_to_code.rst:13]: I treat WSL also as "any Linux"

File: docs/source/developer/how_to_release.rst (hunk 6 above)
Comment (patrikspiess) [reply docs/source/developer/how_to_release.rst]: fixed

File: docs/source/developer/how_to_code.rst (hunk 5 above)
Comment (patrikspiess) [reply docs/source/developer/how_to_code.rst:66]: This is documeted under *How to Release*

File: docs/source/developer/how_to_code.rst (hunk 4 above)
Comment (patrikspiess) [reply docs/source/developer/how_to_code.rst]: fixed

File: docs/source/developer/how_to_code.rst (hunk 3 above)
Comment (patrikspiess) [reply docs/source/developer/how_to_code.rst:56]: You're right. I updated the tox settings

File: docs/source/developer/how_to_code.rst (hunk 1 above)
Comment (lucmurer) [reply docs/source/developer/how_to_code.rst:13]: I thought perhaps we should point this out, but if this is clear to Windows users, its ok for me :-)

**Review by lucmurer — APPROVED** [review]

**Pull request merged**